toolboxes/feel/feelmodels/modelcore/modelphysics.hpp

```diff
@@ -633,11 +664,123 @@ public :
+private :
+    std::set<std::string> M_interfaceFluid, M_interfaceSolid;
+};
+
+
+template <uint16_type Dim>
+class ModelPhysicCoefficientFormPDE : public ModelPhysic<Dim>
+{
+    using super_type = ModelPhysic<Dim>;
+    using self_type = ModelPhysicCoefficientFormPDE<Dim>;
+public :
+    struct Infos
+    {
+        enum class Coefficient { convection=0, diffusion, reaction, firstTimeDerivative, secondTimeDerivative, source, conservativeFluxConvection, conservativeFluxSource, curlCurl };
+
+        using shape_dim_type = std::pair<uint16_type,uint16_type>;
+        using shapes_dim_type = std::vector<shape_dim_type>;
+
+        Infos() = default;
+        Infos( std::string const& name, nl::json const& jarg );
+        Infos( std::string const& name, std::string const& unknownName, std::string const& unknownSymbol, std::string const& unknownBasis )
+            :
+            M_equationName( name ),
+            M_unknownName( unknownName ),
+            M_unknownSymbol( unknownSymbol ),
+            M_unknownBasis( unknownBasis )
+            {}
+        Infos( Infos const& ) = default;
+        Infos( Infos && ) = default;
+        std::string const& equationName() const { return M_equationName; }
+        std::string const& unknownName() const { return M_unknownName; }
+        std::string const& unknownSymbol() const { return M_unknownSymbol; }
+        std::string const& unknownBasis() const { return M_unknownBasis; }
+
+        std::map<Coefficient,std::tuple<std::string,shapes_dim_type>> const& coefficientProperties() const { return M_coefficientProperties; }
+        std::string const& coefficientName( Coefficient c ) const { return std::get<0>( M_coefficientProperties.at( c ) ); }
+    private :
+        std::string M_equationName, M_unknownName, M_unknownSymbol, M_unknownBasis;
+        std::map<Coefficient,std::tuple<std::string,shapes_dim_type>> M_coefficientProperties;
+    };
+    using infos_type = self_type::Infos;
+    using infos_ptrtype = std::shared_ptr<infos_type>;
+
+    ModelPhysicCoefficientFormPDE( ModelPhysics<Dim> const& mphysics, std::string const& modeling, std::string const& type, std::string const& name, ModelModel const& model = ModelModel{} );
+    ModelPhysicCoefficientFormPDE( ModelPhysicCoefficientFormPDE const& ) = default;
+    ModelPhysicCoefficientFormPDE( ModelPhysicCoefficientFormPDE && ) = default;
+
+    std::shared_ptr<infos_type> const& infos() const { return M_infos; }
+
+    using Coefficient = typename infos_type::Coefficient;
+    bool hasCoefficient( Coefficient c ) const { return this->hasCoefficient( M_infos->coefficientName( c ) ); }
+    ModelExpression const& coefficient( Coefficient c ) const { return this->coefficient( M_infos->coefficientName( c ) ); }
+
+    template <int M,int N,typename SymbolsExprType = symbols_expression_empty_t>
+    auto coefficientExpr( Coefficient c, SymbolsExprType const& se = symbols_expression_empty_t{} ) const
```

Comment:
const& ?

Reply:
no, here we build a new object from symbols expr